packages/config/src/__tests__/Project-test.ts

```diff
@@ -0,0 +1,86 @@
+import { getNativeBuildVersion, getRuntimeVersionNullable } from '../Project';
+
+describe(getNativeBuildVersion, () => {
+  const version = '1.0.0';
+  const versionCode = 42;
+  const buildNumber = '13';
+  it('works for android', () => {
+    expect(getNativeBuildVersion({ version, android: { versionCode } }, 'android')).toBe(
+      `${version}(${versionCode})`
+    );
+  });
+  it('works for ios', () => {
+    expect(getNativeBuildVersion({ version, ios: { buildNumber } }, 'ios')).toBe(
+      `${version}(${buildNumber})`
+    );
+  });
+  it('throws an error if platform is not recognized', () => {
+    const fakePlatform = 'doesnotexit';
+    expect(() => {
+      getNativeBuildVersion({ version }, fakePlatform as any);
+    }).toThrow(`"${fakePlatform}" is not a supported platform. Choose either "ios" or "android".`);
+  });
+  it('throws an error if the version is missing', () => {
+    expect(() => {
+      getNativeBuildVersion({}, 'ios');
+    }).toThrow('Missing "version" field');
+  });
+  it('throws an error if the platform is ios and the buildNumber is missing', () => {
+    expect(() => {
+      getNativeBuildVersion({ version }, 'ios');
+    }).toThrow(
+      'The "ios.buildNumber" field is required when computing the native build version for ios.'
+    );
+  });
+  it('throws an error if the platform is android and the  versionCode is missing', () => {
+    expect(() => {
+      getNativeBuildVersion({ version }, 'android');
+    }).toThrow(
+      'The "android.versionCode" field is required when computing the native build version for android.'
+    );
+  });
+});
+describe(getRuntimeVersionNullable, () => {
+  it('works if the top level runtimeVersion is a string', () => {
+    const runtimeVersion = '42';
+    expect(getRuntimeVersionNullable({ runtimeVersion }, 'ios')).toBe(runtimeVersion);
+  });
+  it('works if the platform specific runtimeVersion is a string', () => {
+    const runtimeVersion = '42';
+    expect(getRuntimeVersionNullable({ ios: { runtimeVersion } }, 'ios')).toBe(runtimeVersion);
+  });
+  it('works if the runtimeVersion is a policy', () => {
+    const version = '1';
+    const buildNumber = '2';
+    expect(
+      getRuntimeVersionNullable(
+        { version, runtimeVersion: { policy: 'nativeBuildVersion' }, ios: { buildNumber } },
+        'ios'
+      )
+    ).toBe(`${version}(${buildNumber})`);
+  });
+  it('returns null if no runtime version is found', () => {
```

Comment:
If we do go with a sometimes-null root-level runtimeVersion field, I think we may want to differentiate between projects that don't have a runtimeVersion at all (and therefore should use SDK version) and projects that do but don't have a single root-level runtimeVersion (and you need to look in the platform objects).